helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -501,21 +512,64 @@ private void calculateAndUpdateBaseline(ClusterModel clusterModel, RebalanceAlgo
     _baselineCalcLatency.endMeasuringLatency();
     LOG.info("Global baseline calculation completed and has been persisted into metadata store.");
 
-    if (isBaselineChanged && shouldSchedulePartialRebalance) {
+    if (isBaselineChanged && shouldTriggerMainPipeline) {
       LOG.info("Schedule a new rebalance after the new baseline calculation has finished.");
-      RebalanceUtil.scheduleOnDemandPipeline(clusterName, 0L, false);
+      RebalanceUtil.scheduleOnDemandPipeline(clusterData.getClusterName(), 0L, false);
     }
   }
 
-  private Map<String, ResourceAssignment> partialRebalance(
+  private void partialRebalance(
```

Comment:
General comment on the overall structure, open to discussion -- 
Can we make partial rebalance a standalone class? This `WagedRebalancer` class is getting more and more complicated, and "partial rebalance" is more like a separate module now, it has its own computation logic and execution context.

Reply:
It would be strange to me that "partial rebalance" lives outside of WagedRebalancer, because it will never be meaningful outside of WAGED context. 

I could be biased here since this class was written like this in the beginning.

Reply:
In general, that will be more object oriented and potentially easier for mock up and test. But you are also right, so I'll let you make the call.

Reply:
I gave a short answer here. I think the bigger problem is how we want to design this new class and how it works with the original WagedRebalancer class - that in itself is a big conversation. Right now the state of WagedRebalancer doesn't warrant that redesign, which I think you would agree too.